### PR TITLE
Rename legacy filenames

### PR DESCRIPTION
**Pull request by rbt**
All reasonably supported versions of Raku, due to NativeCall and threading issues, allow the newer filenames. Use them here too.

### CODE REVIEW

File: META6.json [hunk 1]
```diff
@@ -59,8 +58,7 @@
...
   ],
-  "test-depends": [
-    "NativeCall::TypeDiag"
-  ],
-  "version": "0.6.5"
-}
+  "api": 1,
+  "source-url": "https://github.com/raku-community-modules/DBIish.git",
+  "raku": "6.c"
+}
```
Comment (abraxxa) [inline META6.json:64]: Do all 6.c versions support the new extensions? If know the raku spec version doesn't correlate with the raku compiler version but can we somehow ensure a new enough raku compiler version is required?

**Review by abraxxa — COMMENTED** [review]
Do all 6.c versions support the new extensions? If know the raku spec version doesn't correlate with the raku compiler version but

**Comment by jonathanstowe** [comment]
I can't see a particular problem with doing this.  I've changed all my modules and haven't had a single issue raised about it.

File: META6.json (hunk 1 above)
Comment (rbt) [reply META6.json]: 6.c goes back a long way.

While DBIish don't need 6.d syntax, it is the least crashy on on interpreters which support it, so I made that change.

File: META6.json [hunk 2]
```diff
@@ -1,55 +1,54 @@
 {
-  "api": 1,
+  "name": "DBIish",
+  "description": "Database connectivity for Raku",
+  "version": "0.6.4",
+  "perl": "6.c",
```
[the remark this answers is not in the document]
Comment (rbt) [reply META6.json]: Ditched the perl. If something complains then that can be backed out.

**Pull request merged**